be/src/storage/primary_index.cpp

```diff
@@ -298,308 +227,227 @@ class FixSliceHashIndex : public HashIndex {
-        }
-    }
-
-    void erase(const vectorized::Column& pks, DeletesMap* deletes) override {
-        auto* keys = reinterpret_cast<const Slice*>(pks.raw_data());
-        uint32_t size = pks.size();
-        FixSlice<S> prefetch_keys[PREFETCHN];
-        size_t prefetch_hashs[PREFETCHN];
-        for (uint32_t i = 0; i < std::min(size, PREFETCHN); i++) {
-            prefetch_keys[i].assign(keys[i]);
-            prefetch_hashs[i] = FixSliceHash<S>()(prefetch_keys[i]);
-            _map.prefetch_hash(prefetch_hashs[i]);
-        }
-        for (auto i = 0; i < size; i++) {
-            uint32_t pslot = i % PREFETCHN;
-            auto iter = _map.find(prefetch_keys[pslot], prefetch_hashs[pslot]);
-            if (iter != _map.end()) {
-                uint64_t old = iter->second.value;
-                (*deletes)[(uint32_t)(old >> 32)].push_back((uint32_t)(old & 0xffffffff));
-                _map.erase(iter);
+            for (auto i = idx_begin; i < idx_end; i++) {
+                uint32_t pslot = (i - idx_begin) % PREFETCHN;
+                auto iter = _map.find(prefetch_keys[pslot], prefetch_hashes[pslot]);
+                if (iter != _map.end()) {
+                    uint64_t old = iter->second.value;
+                    (*deletes)[(uint32_t)(old >> 32)].push_back((uint32_t)(old & 0xffffffff));
+                    _map.erase(iter);
+                }
+                uint32_t prefetch_i = i + PREFETCHN;
+                if (LIKELY(prefetch_i < idx_end)) {
+                    prefetch_keys[pslot].assign(keys[prefetch_i]);
+                    prefetch_hashes[pslot] = FixSliceHash<S>()(prefetch_keys[pslot]);
+                    _map.prefetch_hash(prefetch_hashes[pslot]);
+                }
             }
-            uint32_t prefetch_i = i + PREFETCHN;
-            if (LIKELY(prefetch_i < size)) {
-                prefetch_keys[pslot].assign(keys[prefetch_i]);
-                prefetch_hashs[pslot] = FixSliceHash<S>()(prefetch_keys[pslot]);
-                _map.prefetch_hash(prefetch_hashs[pslot]);
+        } else {
+            for (auto i = idx_begin; i < idx_end; i++) {
+                auto iter = _map.find(FixSlice<S>(keys[i]));
+                if (iter != _map.end()) {
+                    uint64_t old = iter->second.value;
+                    (*deletes)[(uint32_t)(old >> 32)].push_back((uint32_t)(old & 0xffffffff));
+                    _map.erase(iter);
+                }
             }
         }
     }
 
     std::size_t memory_usage() const final { return _map.capacity() * (1 + S * 4 + sizeof(RowIdPack4)); }
 
     std::string memory_info() const override {
-        auto caps = _map.capacities();
-        string caps_str;
-        for (auto e : caps) {
-            StringAppendF(&caps_str, "%zu,", e);
-        }
-        return Substitute("$0M($1/$2 $3)", memory_usage() / (1024 * 1024), size(), capacity(), caps_str);
+        return Substitute("$0M($1/$2)", memory_usage() / (1024 * 1024), size(), capacity());
```

Comment:
can this info be logged or returned to user?
if so , add more description for the merics to be more readable

Reply:
No, it's just used in benchmark code(not included in this PR), I can remove them.